src/astra_driver.cpp

```diff
@@ -631,28 +931,36 @@ void AstraDriver::newIRFrameCallback(sensor_msgs::ImagePtr image)
 
 void AstraDriver::newColorFrameCallback(sensor_msgs::ImagePtr image)
 {
+  boost::mutex::scoped_lock lock(color_syncer_.mutex);
+
   if ((++data_skip_color_counter_)%data_skip_==0)
   {
     data_skip_color_counter_ = 0;
 
-    if (color_subscribers_)
+    if (color_subscribers_ or color_syncer_.need_image)
     {
       image->header.frame_id = color_frame_id_;
       image->header.stamp = image->header.stamp + color_time_offset_;
 
+      color_syncer_.image = image;
```

Comment:
you will hold an additional image in memory - no madder if you really need it or not. please add it only if you really need a image